astropix.py

```diff
@@ -212,15 +256,37 @@ def init_voltages(self, slot: int, vcal:float, vsupply: float, vthreshold:float,
         inj.pulsesperset = 1
     """
     # Setup Injections
-    def init_injection(self, dac_settings:tuple[int, list[float]] = (2, [0.4, 0.0]), position: int = 3, inj_period:int = 100, clkdiv:int = 400, initdelay: int = 10000, cycle: float = 0, pulseperset: int = 1):
+    def init_injection(self, slot: int = 3, inj_voltage:float = None, inj_period:int = 100, clkdiv:int = 400, initdelay: int = 10000, cycle: float = 0, pulseperset: int = 1, dac_config:tuple[int, list[float]] = (2, [0.4, 0.0])):
+        """
+        Configure injections
+        No required arguments. No returns.
+        Optional Arguments:
+        slot: int - Location of the injection module
+        inj_voltage: float - Injection Voltage. Range from 0 to 1.8. If dac_config is set inj_voltage will be overwritten
+        inj_period: int
+        clkdiv: int
+        initdelay: int
+        cycle: float
+        pulseperset: int
+        dac_config:tuple[int, list[float]]: injdac settings. Must be fully specified if set. 
+        """
+        default_injdac = (2, [0.4, 0.0])
         # Some fault tolerance
         try:
             self._voltages_exist
         except:
-            raise Exception("init_voltages must be called first!")
+            raise RuntimeError("init_voltages must be called first!")
+        # Sets the dac_setup if it isn't specified
+        if dac_config is None:
+            dac_settings = default_injdac
+        else:
+            dac_settings = dac_config
+        # The dac_config takes presedence
+        if inj_voltage is not None and dac_config is None:
```

Comment:
This is redundant - you can either define the injection voltage as `inj_voltage` or as an argument in `dac_config`? Maybe we can eliminate `dac_config` as a necessary argument and make the other elements defaults (once we are confident we know what they are)

Reply:
`dac_config` is default `None` (it was accidentally set before) and the default is stored in a variable. I want it to be so if a user specifies a `dac_config` it takes priority over other voltage arguments. I feel this is the correct balance of user configureability and ease of use/fault tolerance. I am not too concerned about speed in this section since it will only be run once.